### PR TITLE
Prefer triple to double equals

### PR DESCRIPTION
**Pull request by willeastcott**
According to [ESLint](https://eslint.org/docs/rules/eqeqeq):

> It is considered good practice to use the type-safe equality operators === and !== instead of their regular counterparts == and !=.
>
> The reason for this is that == and != do type coercion which follows the rather obscure Abstract Equality Comparison Algorithm. For instance, the following statements are all considered true:
>
> [] == false
> [] == ![]
> 3 == "03"
>
>If one of those occurs in an innocent-looking statement such as a == b the actual problem is very difficult to spot.

This PR does not apply this principle to all occurrences of `==` and `!=`. It leaves those testing against `null`, where sometimes the use was intentional. I still believe those should be reworked, but for now, this PR just deals with the 'safe' instances.

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/resources/parser/glb-parser.js [hunk 1]
```diff
@@ -971,7 +971,7 @@ var createMaterial = function (gltfMaterial, textures, disableFlipV) {
             color = specData.diffuseFactor;
             // Convert from linear space to sRGB space
             material.diffuse.set(Math.pow(color[0], 1 / 2.2), Math.pow(color[1], 1 / 2.2), Math.pow(color[2], 1 / 2.2));
-            material.opacity = (color[3] != null) ? color[3] : 1;
+            material.opacity = color[3];
```
Comment (slimbuck) [inline src/resources/parser/glb-parser.js:974]: was this change intentional?

**Review by slimbuck — COMMENTED** [review]
This PR has the potential to break existing code. Are we ok with that?

File: src/resources/parser/glb-parser.js (hunk 1 above)
Comment (willeastcott) [reply src/resources/parser/glb-parser.js:974]: Yes. The spec states that `diffuseFactor` is a 4-element array. I don't see a circumstance where the fourth element can be `null` or `undefined`. Am I missing something?

**Comment by willeastcott** [comment]
> This PR has the potential to break existing code. Are we ok with that?
Every PR has the potential to break existing code. You have to point out where it actually does break code. I believe it doesn't. 😄

File: src/resources/parser/glb-parser.js (hunk 1 above)
Comment (willeastcott) [reply src/resources/parser/glb-parser.js:974]: See here: https://kcoley.github.io/glTF/extensions/2.0/Khronos/KHR_materials_pbrSpecularGlossiness/#specular---glossiness

Comment (slimbuck) [reply src/resources/parser/glb-parser.js:974]: Makes sense.

**Comment by willeastcott** [comment]
Just thinking about it, unease and swapping `==` with `===` is a great argument for not using `==`! 😆

**Review by slimbuck — COMMENTED** [review]
I was thinking about https://github.com/playcanvas/engine/pull/3044/files#diff-b8e35fe9f6d2040346e21c7bf17fbdd77c192d84ca3714046bb4a690abc108fcR209 - where users could currently be passing in a string value instead of number. Code that currently skips a string set - say newValue='0' and oldValue=0 - will no longer be ignored.

This is a pretty contrived example though so probably worth the risk.

**Review by slimbuck — APPROVED** [review]
